### DISCUSSION TITLE
Cannot install, multiple attempts, same error.

### DISCUSSION THREAD

**Discussion by socksdoxie** [Homebrew/discussions#3227] [Getting started]
Follow directions, but not able to install Homebrew: Error: Failed to link all completions, docs and manpages:
  Permission denied @ rb_file_s_symlink - (../../../Homebrew/completions/zsh/_brew, /usr/local/share/zsh/site-functions/_brew)
Failed during: /usr/local/bin/brew update --force --quiet...Whys is permission denied and by who? I am a single user, my own admin. New to this space, can someone explain. Same error w/ and without VPN. Using MacOS Monteray ver12.

**Comment by gromgit** [comment]
>```
> Permission denied @ rb_file_s_symlink - (../../../Homebrew/completions/zsh/_brew, /usr/local/share/zsh/site-functions/_brew)
>```

What's the output of `ls -ld /usr/local/share/zsh/site-functions`?

**Reply by socksdoxie** [reply]
Sorry just getting back to you....I am new at this not sure I did it right per your email. Ido use a VPN. These are my responses based how i did it each instance:  % ls /usr/local/share/zsh/site-functions
_mullvad
2) I think this is what you are looking for: ls -ld /usr/local/share/zsh/site-functions
drwxr-xr-x  3 root  wheel  96 Mar 11 14:54 /usr/local/share/zsh/site-functions

I was signed in on my user account, not the admin. Does that makes a difference? Thanks

**Reply by gromgit** [reply]
Yes, it entirely explains the "permission denied" error you got. `sudo chown -R $(whoami) /usr/local/share/zsh/site-functions` should fix that, then run the Homebrew installer again to ensure everything is installed correctly.

**Reply by MelvinQc542** [reply]
goodCatch, @gromgit !

**Comment by wenslayer** [comment]
I just helped a developer with this very same problem on a fresh machine build. I suspect this is a default file now created as root, and so everybody installing Brew on an Intel Mac going forward is going to run into this problem.

I wonder if the installer should do some permissions checks (or outright chowns) beforehand?

**Reply by Jopernikus64** [reply]
In that path /usr/local/share/zsh/site-functions on my Mac I found _mdatp -> /Applications/Microsoft Defender.app/Contents/Resources/Tools/mdatp_completion.zsh
So I guess Defender might have done something there during it's installation.

**Comment by roxtaz** [comment]
`sudo chown -R $(whoami) $(brew --prefix)/*` fixed this error for me.
